app/containers/message/helpers/mediaDownload/autoDownloadPreference.ts

```diff
@@ -0,0 +1,27 @@
+import NetInfo, { NetInfoStateType } from '@react-native-community/netinfo';
+
+import {
+	IMAGES_PREFERENCE_DOWNLOAD,
+	AUDIO_PREFERENCE_DOWNLOAD,
+	VIDEO_PREFERENCE_DOWNLOAD,
+	MediaDownloadOption
+} from '../../../../lib/constants';
+import userPreferences from '../../../../lib/methods/userPreferences';
+import { IUser, IUserMessage } from '../../../../definitions';
+
+type TMediaType = typeof IMAGES_PREFERENCE_DOWNLOAD | typeof AUDIO_PREFERENCE_DOWNLOAD | typeof VIDEO_PREFERENCE_DOWNLOAD;
+interface IUsersParam {
+	user: IUser;
+	author?: IUserMessage;
+}
+
+export const isAutoDownloadEnabled = async (mediaType: TMediaType, { author, user }: IUsersParam) => {
+	const mediaDownloadPreference = userPreferences.getString(mediaType);
+	const netInfoState = await NetInfo.fetch();
+
+	return (
+		(mediaDownloadPreference === MediaDownloadOption.WIFI && netInfoState.type === NetInfoStateType.wifi) ||
+		mediaDownloadPreference === MediaDownloadOption.WIFI_MOBILE_DATA ||
+		author?._id === user.id
```

Comment:
You're coupling `author` to a function that should be only related to download preference and net info state.
If we ever need to reuse this function outside of message scope, we'll have to send `author` and `user` and that's going to require a refactor.
Also this function is colocated in the wrong place, even though it's used only on message.
I'd prefer to have it closer to `handleMediaAutoDownload`, because they are tied to each other.

Reply:
> we'll have to send `author` and `user` and that's going to require a refactor.

To avoid repeating the test of author and user at Image and Audio, I added this condition to autoDownloadPreference and refactored the type from obligatory to optional, wdyt?